samples/convert-upload-images-powervs/convert-upload-images-powervs

```diff
@@ -0,0 +1,742 @@
+  local url=$1
+  local image_name=${url##*/}
+  rm -rf $image_name
+  retry "curl -fsSL $url -o ./$image_name"
+  if [[ $? -eq 0 ]]; then
+    #IMAGE_PATH=$(realpath ./$image_name)
+    IMAGE_PATH=./$image_name
+    DOWNLOAD_IMAGE_NAME=$image_name
+  else
+    error "Unable to fetch the url"
+  fi
+}
+
+#-------------------------------------------------------------------------
+# Download the images if needed
+#-------------------------------------------------------------------------
+function download_image {
+  if [[ "$1" == "rhel" ]];then
+    if echo $RHEL_URL | grep -q -i  'access.cdn.redhat.com' ; then
+      log "downloading rhel image"
+      download_url $RHEL_URL
+      RHEL_IMAGE=$IMAGE_PATH
+      RHEL_DOWNLOADED_IMAGE_NAME=$DOWNLOAD_IMAGE_NAME
+      RHEL_NEW_IMAGE_PATH=$IMAGE_NEW_PATH
+      COPY_RHEL_IMAGE=1
+    fi
+  elif [[ "$1" == "rhcos" ]];then
+    download_url $RHCOS_URL
+    RHCOS_IMAGE=IMAGE_PATH
+    RHCOS_DOWNLOAD_IMAGE_NAME=$DOWNLOAD_IMAGE_NAME
+    copy_image_file $RHCOS_IMAGE $RHCOS_OBJECT_NAME
+    RHCOS_NEW_IMAGE_PATH=$IMAGE_NEW_PATH
+    COPY_RHCOS_IMAGE=1
+  else
+    warn "Unknown image"
+  fi
+}
+
+function main {
+  mkdir -p ./logs
+  vars=""
+
+  # Only use sudo if not running as root
+  [ "$(id -u)" -ne 0 ] && SUDO=sudo || SUDO=""
+  platform_checks
+
+  # Parse commands and arguments
+  while [[ $# -gt 0 ]]; do
+    case "$1" in
+    "--rhel-url")
+      shift
+      RHEL_URL="$1"
+      ;;
+    "--rhcos-url")
+      shift
+      RHCOS_URL="$1"
+      ;;
+    "--cos-bucket")
+      shift
+      COS_BUCKET_NAME="$1"
+      ;;
+    "--service-name")
+      shift
+      SERVICE_NAMES="$1"
+      SERVICE_NAME=${SERVICE_NAMES%%,*}
+      SERVICE_NAMES=$(echo $SERVICE_NAMES | sed 's/,/ /g')
+      ;;
+    "--region")
+      shift
+      COS_BUCKET_REGION="$1"
+      ;;
+    "--cos-resource-group")
+      shift
+      COS_RESOURCE_GROUP="$1"
+      ;;
+    "--cos-instance-name")
+      shift
+      COS_INSTANCE_NAME="$1"
+      ;;
+    "--help")
+      help
+      ;;
+    esac
+    shift
+  done
+
+  if [ -z "${COS_BUCKET_NAME}" ] ||  [ -z "${SERVICE_NAME}" ] || [ -z "${COS_BUCKET_REGION}" ]  || [ -z "${COS_RESOURCE_GROUP}" ]; then
+    log "Some or all of the parameters are empty"
+    help
+  fi
+
+  if [ -z "${RHCOS_URL}" ] && [ -z "${RHEL_URL}" ] ; then
+    error "No RHEL/RHCOS/s provided"
+  fi
+  RHEL_SUBS_PASSWORD="${RHEL_SUBSCRIPTION_PASSWORD}"
+  VAR_RHEL_SUBS_USER="${RHEL_SUBSCRIPTION_USERNAME}"
+  setup_image_env_variables
+  setup
+  setup_pvsadm_local
+  variables
+
+  # Downloading rhel locally, else the url may expire by the time the VM comes up
+  if ! [ -z "${RHEL_URL}" ];then
+    SECONDS=0
+    MSG="Downloading rhel qcow2 image completed in"
+    download_image "rhel"
+    time_taken "$MSG"
+  fi
+  create_vm
+  setup_pvsadm_remote
+  prepare_remote
+
+  if ! [ -z "${RHEL_URL}" ];then
+    copy_image_to_remote
+    convert_image_remote "rhel" "${RHEL_IMAGE_NAME}" "${RHEL_URL}"
```

Comment:
What happens when `convert_image_remote` and `upload_image_remote` returns 1? Does the code execute ahead? If yes then check for return value and call `destroy` once here in case the return value is 1 instead of calling destroy from every function.